src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,177 @@
+
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Duke {
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_BYE = "bye";
+    public static final String COMMAND_DONE = "done";
+    public static final String COMMAND_ADD_TODO = "todo";
+    public static final String COMMAND_ADD_DEADLINE = "deadline";
+    public static final String COMMAND_ADD_EVENT = "event";
+    public static final String COMMAND_DELETE = "delete";
+
+    public static final String DUKE_DIVIDER = ">>>>++----------------------------------";
+    public static final String DUKE_GREETINGS = "Hello! I'm Duke!\n"
+            + "What can I do for you?";
+    public static final String DUKE_LOGO = "Hello from\n"
+            + " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    public static final String DUKE_BYE = "Bye. Hope to see you again soon!";
+
+    public static final String ERROR_TODO_NO_DESCRIPTION = "☹ OOPS!!! The description of a Todo cannot be empty.";
+    public static final String ERROR_DEADLINE_NO_DESCRIPTION = "☹ OOPS!!! The description of a Deadline cannot be" +
+            " empty.";
+    public static final String ERROR_EVENT_NO_DESCRIPTION = "☹ OOPS!!! The description of an Event cannot be empty.";
+    public static final String ERROR_NO_DEADLINE = "☹ OOPS!!! Your Deadline doesn't contain a deadline!";
+    public static final String ERROR_NO_EVENT = "☹ OOPS!!! Your Event doesn't contain a date";
+    public static final String ERROR_NO_DONE_ARGUMENT = "☹ OOPS!!! You need an argument to be done with!";
+    public static final String ERROR_DONE_ARGUMENT = "☹ OOPS!!! Your done argument is invalid!";
+    public static final String ERROR_UNKNOWN_COMMAND = "☹ OOPS!!! I'm sorry, but I don't know what that means :-(";
+    public static final String ERROR_TASK_LOAD = "☹ OOPS!!! I'm sorry, but I couldn't load the tasks :-(";
+    public static final String ERROR_TASK_FORMAT = "☹ OOPS!!! Your tasks are in the wrong format :-(";
+    public static final String ERROR_TASK_SAVE = "☹ OOPS!!! I'm sorry, but I couldn't save the tasks :-(";
+
+    private static final TaskManager taskManager = new TaskManager();
+
+    public static void showDivider() {
+        System.out.println(DUKE_DIVIDER);
+    }
+
+    public static void greet() {
+        System.out.println(DUKE_LOGO);
+        showDivider();
+        System.out.println(DUKE_GREETINGS);
+        showDivider();
+    }
+
+    public static void bye() {
+        System.out.println(DUKE_BYE);
+    }
+
+    public static void checkArgumentsLength(int argumentLength, int expectedLength, String errorMessage)
+            throws InvalidArgumentException {
+        if (argumentLength < expectedLength) {
+            throw new InvalidArgumentException(errorMessage);
```

Comment:
Be careful with indentation here

Reply:
According to the Java coding standards indicated in the course materials, shouldn't the wrapped line have 8 more spaces than the parent line?

"Indentation for wrapped lines should be 8 spaces (i.e. twice the normal indentation of 4 spaces) more than the parent line."

https://se-education.org/guides/conventions/java/basic.html#layout

Reply:
Yes my bad. Overlooked that it was a wrapped line in a hurry!